SU2_CFD/include/output/COutputFactory.hpp

```diff
@@ -0,0 +1,69 @@
+/*!
+ * \file COutput.hpp
+ * \brief Headers of the output class.
+ * \author T.Albring
+ * \version 7.0.2 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2020, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "../../../Common/include/option_structure.hpp"
+
+class COutput;
+class CConfig;
+class COutputLegacy;
+
+class COutputFactory{
+
+private:
+
+  /*!
+   * \brief Private constructor to avoid creating instances of this class
+   */
+  COutputFactory();
```

Comment:
Thanks for the const stuff, everything looks very nice.
I think it is a good idea to have a separate numerics PR.

Final note, the preferred C++11 way to achieve this is to delete the default constructor instead of making it private.

Reply:
Perfect, thanks! If that's the preferred way then we should do it :+1:

Reply:
At least according to Scott Meyers xD